storm-core/src/jvm/org/apache/storm/executor/spout/SpoutExecutor.java

```diff
@@ -79,8 +79,11 @@ public SpoutExecutor(final Map workerData, final List<Long> executorId, Map<Stri
         this.spoutThrottlingMetrics = new SpoutThrottlingMetrics();
     }
 
-    @Override
     public void init(final Map<Integer, Task> idToTask) {
+        while (!stormActive.get()) {
```

Comment:
The assumption here is that this method is invoked only once. Do no we want to handle for multiple invocations?